CMakeLists.txt

```diff
@@ -1,41 +1,99 @@
 cmake_minimum_required(VERSION 3.16)
 set(OpenGL_GL_PREFERENCE GLVND)
 
+option(USE_VCPKG "Use vcpkg to handle some of the dependencies" OFF)
+if(DEFINED ENV{VCPKG_ROOT} AND NOT DEFINED CMAKE_TOOLCHAIN_FILE
+    OR USE_VCPKG)
+    set(OBE_USE_VCPKG ON)
+    set(CMAKE_TOOLCHAIN_FILE "$ENV{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake"
+        CACHE STRING "")
+endif()
+
+if(OBE_USE_VCPKG)
+    if(NOT DEFINED ENV{VCPKG_FEATURE_FLAGS})
+        set(ENV{VCPKG_FEATURE_FLAGS} "manifests") # manifests are required because of vcpkg.json
+    endif()
+        
+    if(NOT DEFINED ENV{VCPKG_DEFAULT_TRIPLET})
+        message(FATAL_ERROR
+            "Using vcpkg to build ObEngine without defining ${VCPKG_DEFAULT_TRIPLET}!! Aborting...")
+    endif()
+endif()
+
 project(ObEngine)
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_SOURCE_DIR}/cmake)
 include(engine_git)
 
 add_subdirectory(extlibs/bezier)
-add_subdirectory(extlibs/catch)
+
+if(NOT OBE_USE_VCPKG)
+    add_subdirectory(extlibs/catch)
+
+    # fmtlib is required by spdlog
+    add_subdirectory(extlibs/fmt)
+    add_subdirectory(extlibs/spdlog)
+
+    # lua is required by sol
+    add_subdirectory(extlibs/lua)
+    add_subdirectory(extlibs/sol3)
+    
+    add_subdirectory(extlibs/tinydir)
+endif()
+
 add_subdirectory(extlibs/dynamicLinker)
 add_subdirectory(extlibs/fswrapper)
 
-# fmtlib is required by spdlog
-add_subdirectory(extlibs/fmt)
-add_subdirectory(extlibs/spdlog)
-
 add_subdirectory(extlibs/vili)
 
-# lua is required by sol
-add_subdirectory(extlibs/lua)
-add_subdirectory(extlibs/sol3)
-
 # Audio library
 add_subdirectory(extlibs/soloud)
 
 if (APPLE OR USE_FILESYSTEM_FALLBACK)
-    add_subdirectory(extlibs/tinydir)
     set(USE_FILESYSTEM_FALLBACK TRUE CACHE BOOL "Use filesystem fallback" FORCE)
 endif()
 
 # minizip is required by elzip
 add_subdirectory(extlibs/minizip)
 add_subdirectory(extlibs/elzip)
 
-# SFML 2.5 is minimum version
-find_package(SFML 2.5 COMPONENTS system window graphics network audio REQUIRED)
-# SFML is required by sfe (SFml Extensions)
+# snippet taken from
+# https://github.com/microsoft/vcpkg/blob/master/ports/sfml/usage
+if(OBE_USE_VCPKG)
+    include(FindLua)
+
+    find_package(lua REQUIRED)
+    find_package(sol2 CONFIG REQUIRED)
+    find_package(fmt CONFIG REQUIRED)
+    find_package(spdlog CONFIG REQUIRED)
+    find_package(Catch2 CONFIG REQUIRED)
+    find_path(TINYDIR_INCLUDE_DIRS "tinydir.h")
+
+    find_package(SFML COMPONENTS system window graphics CONFIG REQUIRED)
+
+    include_directories(${TINYDIR_INCLUDE_DIRS})
```

Comment:
Can we remove those now that they have been added into the [`src/Core/CMakeLists.txt`](https://github.com/Sygmei/ObEngine/blob/master/src/Core/CMakeLists.txt) ?